src/main/scala/scala.camp/repository/UserRepository.scala

```diff
@@ -0,0 +1,44 @@
+package scala.camp.repository
+
+import slick.jdbc.H2Profile.api._
+import slick.lifted.Tag
+
+import scala.camp.model.User
+import scala.concurrent.Future
+
+trait UserRepository[F[_]] {
+
+  def registerUser(user: User): F[Int]
+
+  def getById(id: Int): F[Option[User]]
+
+  def getByUsername(username: String): F[Seq[User]]
+}
+
+trait UserRepositoryImpl extends UserRepository[Future] {
+
+  lazy val db = Database.forConfig("database")
+  lazy val userTable = TableQuery[UserTable]
+
+  class UserTable(tag: Tag) extends Table[User](tag, "users") {
+    val id = column[Int]("id", O.PrimaryKey, O.AutoInc)
+    val userName = column[String]("username")
+    val address = column[Option[String]]("address")
+    val email = column[String]("email")
+
+    def * = (id, userName, address, email).mapTo[User]
+  }
+
+  def registerUser(user: User): Future[Int] = {
+    db.run(userTable returning userTable.map(_.id) += user)
+  }
```

Comment:
you should use slightly modified version of retry async which retries some DB SQL exceptions.